source/beamline/AMDetector.cpp

```diff
@@ -428,28 +459,34 @@ bool AMDetector::clear(){
 	return clearImplementation();
 }
 
-void AMDetector::setAsDarkCurrentMeasurementValue(){
+void AMDetector::setLastMeasurementAsDarkCurrent()
+{
+	if (canDoDarkCurrentCorrection()) {
+		double newTime = acquisitionTime();
+		double newValue = double(singleReading()) / newTime;
+
+		updateDarkCurrentTime(newTime);
+		updateDarkCurrentValue(newValue);
+		updateDarkCurrentValidState(true);
```

Comment:
I have public setters already for setDarkCurrentValue(), setDarkCurrentTime(), and setDarkCurrentValidState()--they are the ones that perform the 'can do dark current?' test and then call the protected 'update' methods that actually perform the update. Is the problem that they should be squeezed into one method?